src/lib/utils/Identity.ts

```diff
@@ -0,0 +1,111 @@
+import dateFormat from "dateformat";
+import fs from "fs";
+import path from "path";
+import untildify from "untildify";
+import { IFingerprint, IFingerprintRequest } from "../../types";
+import { Base64 } from "./Base64";
+import { sha256 } from "./crypto";
+import { Fingerprint } from "./Fingerprint";
+import { randomIntFromInterval, randomString, randomString2 } from "./strings";
+
+/**
+ * Used to load Fingerprint
+ *
+ * @public
+ */
+export class Identity {
+    private _filepath: string;
+    private _fingerprint: Fingerprint;
+
+    constructor(filePath: string) {
+        this._filepath = path.resolve(untildify(path.normalize(filePath)));
+
+        // load fingerprint from file
+        const txt = fs.readFileSync(this._filepath).toString();
+        const fpdata = JSON.parse(txt);
+        this._fingerprint = new Fingerprint(fpdata);
+        this.update(); // shuffle data
+    }
+
+    /**
+     * Overwrite request data in fingerprint
+     */
+    public setRequest(data: IFingerprintRequest): void {
+        this._fingerprint.setRequest(data);
+    }
+
+    /**
+     * Shuffle data in fingerprint
+     */
+    public async update(): Promise<void> {
+        await this._fingerprint.shuffle();
+    }
+
+    /**
+     * @return Fingerprint json object
+     */
+    public get fingerprint(): IFingerprint {
+        return this._fingerprint.fingerprint;
+    }
+
+    /**
+     * Save fingerprint to file
+     */
+    public save(): void {
+        fs.writeFileSync(this._filepath, JSON.stringify(this._fingerprint.fingerprint));
+    }
+
+    /**
+     * Generate new Identity
+     * Values are random generated, and can be incorrect (can be detectable and you could be banned)
+     */
+    public static generateIdentity(filePath: string): void {
```

Comment:
function is public so that the user can generate a new identity.
Comment about incorrect values is because I random generate all random string, hashes etc.
Auth works but uses should manually change for example GPU name etc

Reply:
To be honest I think instead of generating the identity with random values, we should expose methods that allow users to decode it from a request made by the client.